### PR TITLE
Remove specific DetIds in reco/geometry ESProducts

### PR DESCRIPTION
**Pull request by pieterdavid**
was: ￼ MeasurementTracker and/or TrackerGeometry available.
See below for details, it comes down to passing on the TrackerTopology pointer from the ESProducer

### CODE REVIEW

File: RecoParticleFlow/PFTracking/interface/PFCheckHitPattern.h [hunk 1]
```diff
@@ -48,8 +48,8 @@ class PFCheckHitPattern {
   /// on track.
 
   PFTrackHitFullInfo 
-    analyze(edm::ESHandle<TrackerGeometry>, const reco::TrackBaseRef track, 
-	    const TransientVertex& vert);
+    analyze(const TrackerTopology* tkerTopo, const TrackerGeometry* tkerGeom,
+            const reco::TrackBaseRef track, const TransientVertex& vert);
```
Comment (OlivierBondu) [inline RecoParticleFlow/PFTracking/interface/PFCheckHitPattern.h:52]: why the change over the `ESHandle` ? This is recommended somewhere so this is free cleaning ?

File: RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc [hunk 2]
```diff
@@ -141,6 +142,7 @@ MeasurementTrackerESProducer::produce(const CkfComponentsRecord& iRecord)
 							          pixelCPE.product(),
 							          stripCPE.product(),
 							          hitMatcher.product(),
+                                                                  trackerTopology.product(),
```
Comment (OlivierBondu) [inline RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: space / tab alignment not matching the original

File: RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc [hunk 3]
```diff
@@ -156,6 +158,7 @@ MeasurementTrackerESProducer::produce(const CkfComponentsRecord& iRecord)
...
```
Comment (OlivierBondu) [inline RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: same here

File: RecoTracker/SiTrackerMRHTools/interface/SimpleDAFHitCollector.h [hunk 4]
```diff
@@ -16,11 +16,12 @@ class StripRecHit1D;
...
 class SimpleDAFHitCollector :public MultiRecHitCollector {
 	public:
-	explicit SimpleDAFHitCollector(const MeasurementTracker* measurementTracker,
+	explicit SimpleDAFHitCollector(const TrackerTopology* trackerTopology,
+                                 const MeasurementTracker* measurementTracker,
 				 const SiTrackerMultiRecHitUpdator* updator,
 			         const MeasurementEstimator* est,
 				 const Propagator* propagator, bool debug
-				 ):MultiRecHitCollector(measurementTracker), theUpdator(updator), theEstimator(est), thePropagator(propagator), debug_(debug){
+				 ):MultiRecHitCollector(measurementTracker), theTopology(trackerTopology), theUpdator(updator), theEstimator(est), thePropagator(propagator), debug_(debug){
```
Comment (OlivierBondu) [inline RecoTracker/SiTrackerMRHTools/interface/SimpleDAFHitCollector.h:24]: shouldn't `theTopology` be the first one to be initialized, to match the (style of the) function definition?

**Review by OlivierBondu — CHANGES_REQUESTED** [review]
Thanks! only very minor comments and questions, looks good to me!

File: RecoTracker/SiTrackerMRHTools/interface/SimpleDAFHitCollector.h (hunk 4 above)
Comment (pieterdavid) [reply RecoTracker/SiTrackerMRHTools/interface/SimpleDAFHitCollector.h:24]: I think it's better style to first initialize the base classes (even if technically it shouldn't matter). I made `TrackerTopology` the first argument because it is a more 'basic' thing than the other arguments.

File: RecoParticleFlow/PFTracking/interface/PFCheckHitPattern.h (hunk 1 above)
Comment (pieterdavid) [reply RecoParticleFlow/PFTracking/interface/PFCheckHitPattern.h:52]: I'm not sure there's a clear recommendation, but I understood they are essentially equivalent (`edm::ESHandle` throws an error on access when the product can't be found, if I read http://cmsdoxygen.web.cern.ch/cmsdoxygen/CMSSW_9_2_0_patch3/doc/html/d0/d69/FWCore_2Framework_2interface_2ESHandle_8h_source.html correctly, so passing the raw pointer only makes this exception show up earlier) - I personally prefer the raw pointer because it makes the interface lighter (and clearer), and I wanted to be consistent, so changed the `TrackerGeometry` handle as well - but I may have missed something.

**Comment by pieterdavid** [comment]
I pushed a fix for the whitespace issues (these files are full of tabs, they predate this year's stackoverflow survey :wink:).

**Review by OlivierBondu — APPROVED** [review]

**Comment by OlivierBondu** [comment]
thanks!
